libhb: fix availability of left/right only mixdowns.

Rodeo314 opened the pull request:
Mixdowns introduced in 8d9969febdcec59fee453888ed0bb6a64adf7679

Availability was (probably accidentally) extended to supersets of Stereo input in e7b1636fca20d91241ff3e1c12919a3123d05daf

        /*
         * The following mixdowns have a very specific purpose!!!
         *
         * Given 2-channel input, they allow discarding either of the right or
         * left channel to produce mono output instead of downmixing (the latter
         * may clip or lower volume depending on whether the downmix coefficients
         * are normalized or not). They are meant to be used for sources which
         * are known to actually be Mono (identical content in the L/R channels
         * or one of L/R is actually empty). They hardly make any sense when the
         * input has more than two channels.
         *
         * Thus specifically only allow them for Stereo input (layout == STEREO).
         */
        case HB_AMIXDOWN_LEFT:
        case HB_AMIXDOWN_RIGHT:
            return (layout == AV_CH_LAYOUT_STEREO);

twalker314 committed:
libhb: fix availability of left/right only mixdowns.

Mixdowns introduced in 8d9969febdcec59fee453888ed0bb6a64adf7679
Availability was (probably accidentally) extended to supersets
of Stereo input in e7b1636fca20d91241ff3e1c12919a3123d05daf

ghost commented:
@jstebbins I think I brought this up on IRC when I originally discovered it, but I cannot remember exactly what came of it (if you had a specific reason to enable the mixdowns for more-than-Stereo input or if you just asked me to fix it and I just never got around to it in, erm, about 5 years).

twalker314 committed:
libhb: no sample rate conversion for Mono Left/Right only mixdowns.

Easier than properly fixing the rather dirty
hack that's used to implement said mixdowns.

Works around issue #3533

ghost commented:
Added a commit to work around issue #3533 by forcing output samplerate to input samplerate.

(how) do we typically reference github issues in commit messages: should I use the full URL, just the issue number, or not mention it in the commit message at all?

galad87 commented:
The issue number is enough imho.
